eth-amp-analytics-wp.php

```diff
@@ -0,0 +1,225 @@
+<?php
+/*
+Plugin Name: Google Analytics for AMP
+Plugin URI: https://ethitter.com/plugins/
+Description: Enable Google Analytics support for Automattic's AMP plugin
+Author: Erick Hitter
+Version: 0.1
+Author URI: https://ethitter.com/
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+class ETH_AMP_Analytics_WP {
+	/**
+	 * PLUGIN SETUP
+	 */
+
+	/**
+	 * Singleton
+	 */
+	private static $instance = null;
+
+	/**
+	 * Instantiate singleton
+	 */
+	public static function get_instance() {
+		if ( ! is_a( self::$instance, __CLASS__ ) ) {
+			self::$instance = new self;
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Dummy magic methods
+	 */
+	public function __clone() {_doing_it_wrong( __FUNCTION__, __( 'Cheatin&#8217; huh?', 'eth-amp-analytics-wp' ), '0.1' ); }
+	public function __wakeup() { _doing_it_wrong( __FUNCTION__, __( 'Cheatin&#8217; huh?', 'eth-amp-analytics-wp' ), '0.1' ); }
+	public function __call( $name = '', $args = array() ) { unset( $name, $args ); return null; }
+
+	/**
+	 * Class properties
+	 */
+	private $plugin_option_name = 'eth_amp_analytics_wp';
+	private $plugin_option_defaults = array(
+		'property_id' => null,
+	);
+
+	private $options = null;
+
+	/**
+	 * Defer plugin action loading until all plugins are loaded
+	 *
+	 * @return null
+	 */
+	private function __construct() {
+		add_action( 'plugins_loaded', array( $this, 'action_plugins_loaded' ) );
+	}
+
+	/**
+	 * Register actions and filters
+	 */
+	public function action_plugins_loaded() {
+		if ( ! defined( 'AMP__FILE__' ) ) {
+			return;
+		}
+
+		// Front-end
+		add_action( 'wp_loaded', array( $this, 'action_wp_loaded' ) );
+
+		// Admin UI
+		add_action( 'admin_init', array( $this, 'action_admin_init' ) );
+		register_uninstall_hook( __FILE__, array( 'ETH_AMP_Analytics_WP', 'uninstall' ) );
+	}
+
+	/**
+	 * Clean up plugin data when it's removed
+	 */
+	public static function uninstall() {
+		delete_option( 'eth_amp_analytics_wp' );
+	}
+
+	/**
+	 * Conditionally load front-end hooks
+	 */
+	public function action_wp_loaded() {
+		if ( ! empty( $this->get_option( 'property_id' ) ) ) {
+			add_action( 'amp_post_template_head', array( $this, 'action_amp_post_template_head' ), 5 ); // Must appear before the AMP JS library
```

Comment:
It'd be better to hook into `post_template_data` and append to the scripts rather than manually outputting. This way, we avoid duplicates of the am-analytics script (e.g. other things may need amp-analytics as well).